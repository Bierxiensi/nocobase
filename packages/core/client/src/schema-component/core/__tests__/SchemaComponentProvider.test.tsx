import React from 'react';
import { render } from '@testing-library/react';
import { connect } from '@formily/react';

import { SchemaComponentProvider } from '../SchemaComponentProvider';
import { SchemaComponent } from '../SchemaComponent';

function App() {
  const ComponentA = connect((props) => {
    return <div>{props.title}</div>;
  });
  const schema = {
    type: 'object',
    properties: {
      componentA: {
        type: 'void',
        'x-component': 'ComponentA',
        'x-component-props': {
          title: 'hello',
        },
      },
    },
  };
  return (
    <SchemaComponentProvider components={{ ComponentA }}>
      <SchemaComponent schema={schema} />
    </SchemaComponentProvider>
  );
}
describe('SchemaComponentProvider', () => {
  it('SchemaComponent get component by SchemaComponentProvider', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
