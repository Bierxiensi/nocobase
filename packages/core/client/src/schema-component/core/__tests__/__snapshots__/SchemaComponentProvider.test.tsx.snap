// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaComponentProvider SchemaComponent get component by SchemaComponentProvider 1`] = `
<div>
  <div>
    hello
  </div>
</div>
`;
